react/utils/Images.js

```diff
@@ -2,12 +2,12 @@ const DEFAULT_WIDTH = 50
 const DEFAULT_HEIGHT = 50
 const DEFAULT_HDF = 1
 
-const baseUrlRegex = new RegExp(/.+ids\/(\d+)(?:-(\d+)-(\d+)|)\//)
+const baseUrlRegex = new RegExp(/.+ids\/((\d+)[a-zA-Z]*)+(?:-(\d+)-(\d+)|)\//)
```

Comment:
You are creating a new capturing group here, which may cause the logic in lines 11-24 to break.
```ts
const oldRegex = new RegExp(/.+ids\/(\d+)(?:-(\d+)-(\d+)|)\//)
const newRegex = new RegExp(/.+ids\/((\d+)[a-zA-Z]*)+(?:-(\d+)-(\d+)|)\//)

oldRegex.exec('fooids/123-456-789/')
// (4) ["fooids/123-456-789/", "123", "456", "789", index: 0, input: "fooids/123-456-789/", groups: undefined]

newRegex.exec('fooids/123-456-789/')
// (5) ["fooids/123-456-789/", "123", "123", "456", "789", index: 0, input: "fooids/123-456-789/", groups: undefined]
```

How about
```suggestion
const baseUrlRegex = new RegExp(/.+ids\/([a-zA-Z0-9]+)(?:-(\d+)-(\d+)|)\//)
```

```ts
const suggestionRegex = new RegExp(/.+ids\/([a-zA-Z0-9]+)(?:-(\d+)-(\d+)|)\//)
suggestionRegex.exec('fooids/123-456-789/')
// (4) ["fooids/123-456-789/", "123", "456", "789", index: 0, input: "fooids/123-456-789/", groups: undefined]
```

Reply:
Nice! I didn't see that. Thanks!

Reply:
I think you can also use something like `(?:.*)` to create a group without capturing